sql/src/test/java/org/apache/druid/sql/calcite/CalciteJoinQueryTest.java

```diff
@@ -5652,4 +5652,128 @@ public void testJoinsWithThreeConditions()
+                                                  .context(context)
+                                                  .build()
+                                        ),
+                                        "j0.",
+                                        "1",
+                                        JoinType.INNER
+                                    ),
+                                    new QueryDataSource(
+                                        GroupByQuery.builder()
+                                                    .setInterval(querySegmentSpec(Filtration.eternity()))
+                                                    .setGranularity(Granularities.ALL)
+                                                    .setDataSource(new TableDataSource(CalciteTests.DATASOURCE1))
+                                                    .setVirtualColumns(expressionVirtualColumn(
+                                                        "v0",
+                                                        "1",
+                                                        ColumnType.LONG
+                                                    ))
+                                                    .setDimensions(
+                                                        new DefaultDimensionSpec("m1", "d0", ColumnType.FLOAT),
+                                                        new DefaultDimensionSpec("v0", "d1", ColumnType.LONG)
+                                                    )
+                                                    .build()
+                                    ),
+                                    "_j0.",
+                                    "(floor(100) == \"_j0.d0\")",
+                                    JoinType.LEFT
+                                )
+                            )
+                            .granularity(Granularities.ALL)
+                            .aggregators(aggregators(
+                                new FilteredAggregatorFactory(
+                                    new CountAggregatorFactory("a0"),
+                                    or(
+                                        new SelectorDimFilter("j0.a0", "0", null),
+                                        and(
+                                            selector("_j0.d1", null, null),
+                                            expressionFilter("(\"j0.a1\" >= \"j0.a0\")")
+                                        )
+
+                                    )
+                                )
+                            ))
+                            .context(getTimeseriesContextWithFloorTime(TIMESERIES_CONTEXT_BY_GRAN, "d0"))
+                            .intervals(querySegmentSpec(Filtration.eternity()))
+                            .context(context)
+                            .build();
+
+    }
+
+    testQuery(
+        "SELECT COUNT(*) FILTER (WHERE FLOOR(100) NOT IN (SELECT m1 FROM foo)) "
+        + "FROM foo",
+        context,
```

Comment:
Should we add a test with join in the query context as "Sort merge", but because of the query structure, msq instead uses broadcast joins?

Reply:
This is a test in that sense since this class is subclassed to execute with both the joins, however, there isn't any assertion made (since it stems from the parent test class). 
I'll add a similar test in the MSQ engine and see if I can add some assertions on the algorithm used.